loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,111 @@
+
+const NAME = "KHR_mesh_instancing";
+
+interface IKHRMeshInstancing {
+    mesh?: number;
+    attributes: { [name: string]: number };
+}
+
+/**
+ * [Proposed Specification](https://github.com/KhronosGroup/glTF/pull/1691)
+ * [Playground Sample](//TODO)
+ * !!! Experimental Extension Subject to Changes !!!
+ */
+export class KHR_mesh_instancing implements IGLTFLoaderExtension {
+    /**
+     * The name of this extension.
+     */
+    public readonly name = NAME;
+
+    /**
+     * Defines whether this extension is enabled.
+     */
+    public enabled: boolean;
+
+    private _loader: GLTFLoader;
+
+    /** @hidden */
+    constructor(loader: GLTFLoader) {
+        this._loader = loader;
+        this.enabled = this._loader.isExtensionUsed(NAME);
+    }
+
+    /** @hidden */
+    public dispose() {
+        delete this._loader;
+    }
+
+    /** @hidden */
+    public loadNodeAsync(context: string, node: INode, assign: (babylonTransformNode: TransformNode) => void): Nullable<Promise<TransformNode>> {
+        return GLTFLoader.LoadExtensionAsync<IKHRMeshInstancing, TransformNode>(context, node, this.name, (extensionContext, extension) => {
+            return this._loader.loadNodeAsync(`#/nodes/${node.index}`, node, (babylonTransformNode) => {
+                const promises = new Array<Promise<any>>();
+                const loadAttribute = (attribute: string, assignBufferFunc: (data: Float32Array, accessor: IAccessor) => void) => {
+                    if (extension.attributes[attribute] == undefined) {
+                        return;
+                    }
+                    const accessor = ArrayItem.Get(`${extensionContext}/attributes/${attribute}`, this._loader.gltf.accessors, extension.attributes[attribute]);
+                    promises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        assignBufferFunc(data, accessor);
+                    }));
+                };
+                let translationBuffer: Nullable<Float32Array> = null;
+                let rotationBuffer: Nullable<Float32Array> = null;
+                let scaleBuffer: Nullable<Float32Array> = null;
+                let translationBufferAccessor: Nullable<IAccessor> = null;
+                let rotationBufferAccessor: Nullable<IAccessor> = null;
+                let scaleBufferAccessor: Nullable<IAccessor> = null;
+
+                loadAttribute("TRANSLATION", (data, accessor) => { translationBuffer = data; translationBufferAccessor = accessor; });
+                loadAttribute("ROTATION", (data, accessor) => { rotationBuffer = data; rotationBufferAccessor = accessor; });
+                loadAttribute("SCALE", (data, accessor) => { scaleBuffer = data; scaleBufferAccessor = accessor; });
+
+                return Promise.all(promises).then(() => {
+                    const anyInstanceAccessor = translationBufferAccessor || rotationBufferAccessor || scaleBufferAccessor;
```

Comment:
Chatted offline. We should probably check to make sure the accessor counts are all the same. One option is to do it in the loadAttribute function and save the count so it can be used here.

Reply:
accessor length error checking in loadAttribute.